packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -146,23 +160,23 @@ class AnalyzeCommand extends FlutterCommand {
       //dev/manual_tests/*/ as package
       //dev/manual_tests/*/ as files
 
-      Directory subdirectory;
-
-      subdirectory = new Directory(path.join(ArtifactStore.flutterRoot, 'packages'));
-      if (subdirectory.existsSync()) {
-        for (FileSystemEntity entry in subdirectory.listSync()) {
-          if (entry is Directory)
-            _addPackage(entry.path, dartFiles, pubSpecDirectories);
+      /// Adds all packages in [subPath], assuming a flat directory structure,
+      /// i.e. each direct child of [subPath] is a plain Dart package.
+      void addFlatPackageList(String subPath) {
```

Comment:
Please don't use nested functions unless strictly necessary, we've had some bad experiences with them on this project. This function should probably just be next to _addPackage.